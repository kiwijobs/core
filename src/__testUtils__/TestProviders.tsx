import React, { FC, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback="mocked loading...">{children}</Suspense>
    </ThemeProvider>
  );
};

export const wrapper = TestProviders; // React Testing Library friendly name.
