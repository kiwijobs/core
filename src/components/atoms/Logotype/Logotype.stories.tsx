import React from 'react';
import { storiesOf } from '@storybook/react';
import { Box } from '../../quarks';
import { Logotype } from './Logotype';

storiesOf('Atoms/Logotype', module).add('Various', () => (
  <>
    <Box display="inline-block" p={3}>
      <Logotype src="https://placebear.com/200/200" />
    </Box>
    <Box display="inline-block" p={3}>
      <Logotype size="60px" src="https://placebear.com/201/200" />
    </Box>
    <Box display="inline-block" p={3}>
      <Logotype size="72px" src="https://placebear.com/202/200" />
    </Box>
    <Box display="inline-block" p={3}>
      <Logotype size="80px" src="https://placebear.com/203/300" />
    </Box>
  </>
));
