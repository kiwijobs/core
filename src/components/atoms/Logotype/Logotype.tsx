import React, { forwardRef, RefForwardingComponent } from 'react';
import { Image, ImageProps } from '../../quarks';

export const Logotype: RefForwardingComponent<HTMLImageElement, ImageProps> = forwardRef(
  ({ sx, ...props }, ref) => (
    <Image
      ref={ref}
      sx={{
        borderRadius: 1,
        border: 1,
        p: 1,
        size: '48px',
        bg: 'white',
        objectFit: 'contain',
        ...sx,
      }}
      {...props}
    />
  )
);
