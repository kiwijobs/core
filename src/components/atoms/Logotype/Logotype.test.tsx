import React from 'react';
import { wrapper } from '../../../__testUtils__';
import { Logotype } from './Logotype';
import { render } from '@testing-library/react';

describe('Logotype', () => {
  it('Matches snapshot', () => {
    expect(render(<Logotype src="/img.jpg" />, { wrapper }).container).toMatchSnapshot();
    expect(
      render(<Logotype src="/img.jpg" size="32px" />, { wrapper }).container
    ).toMatchSnapshot();
    expect(render(<Logotype src="/img.jpg" size={60} />, { wrapper }).container).toMatchSnapshot();
  });
});
