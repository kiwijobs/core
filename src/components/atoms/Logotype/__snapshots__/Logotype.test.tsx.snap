// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logotype Matches snapshot 1`] = `
.c0 {
  border-radius: 0.3rem;
  border: 1px solid rgb(228,230,237);
  padding: 4px;
  width: 48px;
  height: 48px;
  background-color: rgb(255,255,255);
  object-fit: contain;
}

<div>
  <img
    class="c0"
    src="/img.jpg"
  />
</div>
`;

exports[`Logotype Matches snapshot 2`] = `
.c0 {
  border-radius: 0.3rem;
  border: 1px solid rgb(228,230,237);
  padding: 4px;
  width: 48px;
  height: 48px;
  background-color: rgb(255,255,255);
  object-fit: contain;
  width: 32px;
  height: 32px;
}

<div>
  <img
    class="c0"
    src="/img.jpg"
  />
</div>
`;

exports[`Logotype Matches snapshot 3`] = `
.c0 {
  border-radius: 0.3rem;
  border: 1px solid rgb(228,230,237);
  padding: 4px;
  width: 48px;
  height: 48px;
  background-color: rgb(255,255,255);
  object-fit: contain;
  width: 60px;
  height: 60px;
}

<div>
  <img
    class="c0"
    size="60"
    src="/img.jpg"
  />
</div>
`;
